### PR TITLE
Fix smaller issues with invalid placeholder type errors

### PR DESCRIPTION
**Pull request by estebank**
Follow up to #70294.

- Fix placement of suggested generic param when bounds are present.
- Reduce error duplication for invalid placeholder types in `fn` types.

r? @Centril

### CODE REVIEW

File: src/librustc_typeck/astconv.rs [hunk 1]
```diff
@@ -2883,10 +2883,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
...
-        if !self.allow_ty_infer() {
+        if !self.allow_ty_infer() && ident_span.is_some() {
             // We always collect the spans for placeholder types when evaluating `fn`s, but we
             // only want to emit an error complaining about them if infer types (`_`) are not
-            // allowed. `allow_ty_infer` gates this behavior.
+            // allowed. `allow_ty_infer` gates this behavior. We check for the presence of
+            // `ident_span` to not emit an error twice when we have `fn foo(_: fn() -> _)`.
             crate::collect::placeholder_type_error(
                 tcx,
                 ident_span.map(|sp| sp.shrink_to_hi()).unwrap_or(DUMMY_SP),
```
Comment (Centril) [inline src/librustc_typeck/astconv.rs]: Can unwrap now and the last argument can be `true`.

**Comment by Centril** [comment]
r=me with ^-- addressed.

**Comment by estebank** [comment]
@bors r=Centril

Nominating for backport (but with much lower priority) as I feel this goes in tandem with #70294.

**Comment by pnkfelix** [comment]
Discussed in [compiler meeting](https://zulip-archive.rust-lang.org/131828tcompiler/40721weeklymeeting2020032654818.html#191889442). Declined for beta backport.

**Pull request merged**